app/Services/StandService.php

```diff
@@ -0,0 +1,131 @@
+<?php
+
+namespace App\Services;
+
+use App\Events\StandAssignedEvent;
+use App\Events\StandUnassignedEvent;
+use App\Exceptions\Stand\StandAlreadyAssignedException;
+use App\Exceptions\Stand\StandNotFoundException;
+use App\Models\Airfield\Airfield;
+use App\Models\Stand\Stand;
+use App\Models\Stand\StandAssignment;
+use App\Models\Vatsim\NetworkAircraft;
+use Illuminate\Database\Eloquent\Builder;
+use Illuminate\Support\Collection;
+
+class StandService
+{
+    public function getStandsDependency(): Collection
+    {
+        return Stand::all()->groupBy('airfield_id')->mapWithKeys(function (Collection $collection) {
+            return [
+                Airfield::find($collection->first()->airfield_id)->code => $collection->map(function (Stand $stand) {
+                    return [
+                        'id' => $stand->id,
+                        'identifier' => $stand->identifier
+                    ];
+                }),
+            ];
+        })->toBase();
+    }
+
+    public function getStandAssignments(): Collection
+    {
+        return StandAssignment::all()->map(function (StandAssignment $assignment) {
+            return [
+                'callsign' => $assignment->callsign,
+                'stand_id' => $assignment->stand_id,
+            ];
+        })->toBase();
+    }
+
+    /**
+     * Creates a stand assignment by assigning an aircraft to a stand.
+     * If the stand is already occupied, then the previous assignment is not
+     * overridden.
+     *
+     * @param string $callsign
+     * @param int $standId
+     * @throws StandAlreadyAssignedException
+     */
+    public function assignAircraftToStand(string $callsign, int $standId): void
+    {
+        $this->getStandById($standId);
+        NetworkDataService::firstOrCreateNetworkAircraft($callsign);
+        $currentAssignment = StandAssignment::where('stand_id', $standId)->first();
```

Comment:
Might be strange, but could you do a findOrFail, catch the model not found exception and do the assignment that way, on the case the assignment doesn't already exist? Would it simplify it at all?

Reply:
I'm a bit on the fence with "Exceptions as flow" - in this case, an assignment not existing isn't an error situation, so I prefer to keep exceptions out of it :) If it were a problem (such as above with stand not existing), I'd probably consider it in that situation

Reply:
Can you tell I've written too much Python recently :D